File: python/leetcode/66. Plus One.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 0
        start = len(digits) - 1

        for i in range(start, -1, -1):
            if i == start:
                digits[start] += 1
            else:
                digits[i] += carry

            if digits[i] < 10:
                return digits

            digits[i] %= 10
            carry = 1

            if i == 0 and carry is not 0:
                digits.insert(0, carry)
        return digits

File: c++/leetcode/66. Plus One.cpp
class Solution
{
public:
  vector<int> plusOne(vector<int> &digits)
  {
    int carry = 0;
    int start = digits.size() - 1;

    for (int i = start; i >= 0; i--)
    {
      if (i == start)
      {
        digits[start] += 1;
      }
      else
      {
        digits[i] += carry;
      }

      if (digits[i] < 10)
        return digits;
      digits[i] %= 10;
      carry = 1;

      if (i == 0 && carry)
      {
        digits.insert(digits.begin(), carry);
      }
    }
    return digits;
  }
};